prow/jira/jira.go

```diff
@@ -364,17 +679,51 @@ func maskAuthorizationHeader(key string, value string) string {
 	return value
 }
 
-// JiraError collapses cryptic Jira errors to include response
+type JiraError struct {
+	statusCode    int
+	body          string
+	originalError error
+}
+
+func (e JiraError) Error() string {
+	return fmt.Sprintf("%s: %s", e.originalError, e.body)
+}
+
+// JiraErrorStatusCode will identify if an error is a JiraError and return the
+// stored status code if it is; if it is not, `-1` will be returned
+func JiraErrorStatusCode(err error) int {
+	jiraErr, ok := err.(*JiraError)
```

Comment:
Please use the `errors` package so this will work with wrapped errors:
```suggestion
        if jiraErr := JiraError{}; errors.As(err, &jiraErr) {
           return jiraErr.statusCode
        }
	jiraErr, ok := err.(*JiraError)
```